packages/chip/src/Chip/Chip.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+
+import { cx } from '@leafygreen-ui/emotion';
+import InlineDefinition from '@leafygreen-ui/inline-definition';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+
+import { DismissButton } from '../DismissButton';
+import { getTruncatedName } from '../utils/getTruncatedName';
+
+import {
+  chipInlineDefinitionClassName,
+  chipTextClassName,
+  chipTextDisabledStyles,
+  chipTextDismissSizeStyle,
+  chipTextSizeStyle,
+  chipTextStyles,
+  chipWrapperBaseDisabledStyles,
+  chipWrapperBaseStyle,
+  chipWrapperDisabledStyle,
+  chipWrapperSizeStyle,
+  chipWrapperThemeStyle,
+} from './Chip.styles';
+import { ChipProps, Size, TruncationLocation, Variant } from './Chip.types';
+
+export const Chip = React.forwardRef<HTMLSpanElement, ChipProps>(
+  (
+    {
+      chipTruncationLocation = TruncationLocation.none,
+      chipCharacterLimit = 15,
+      disabled = false,
+      size = Size.Default,
+      variant = Variant.Blue,
+      darkMode: darkModeProp,
+      label,
+      onDismiss,
+      popoverZIndex,
+      className,
+      dismissButtonAriaLabel,
+      ...rest
+    }: ChipProps,
+    forwardedRef,
+  ) => {
+    const { darkMode, theme } = useDarkMode(darkModeProp);
+
+    const isTruncated =
```

Comment:
these first two `!!` will never not be true bc you are providing default values. We can definitely clean up this logic.

Reply:
I actually think I also disagree with the below. I don't think we should block on chipTruncationLocation, I think setting a chipCharacterLImit should be enough. We should have a fallback chipTruncationLocation, but double confirm this with design

Reply:
I think that makes sense. Technically right now, you can either set `none` or set the limit to `0` to prevent truncation. We shouldn't have two ways to prevent truncation.

Reply:
@alvenw did ask for the default to have no truncation. We can set the default `chipCharacterLImit` to `0` and make the default `chipTruncationLocation` `end`. 

This would be a major change for `Combobox` since the API would be different.

Reply:
Ah - interesting design decision on the Combobox side. Would get sign off from whoever wrote that API in the first place to ensure they weren't thinking through an edge case I'm missing. If not, I'm happy to set defaults that mimic the current behavior, but still modify the API. Alternatively, we can leave things as is in Combobox, but make the change here in the standalone Chip.

Reply:
@TheSonOfThomp Is there an edge case we're not seeing?

Reply:
We chatted about this in person but when the limit is 0 the chip should still truncate, so we need to update the logic to check for `undefined` not `!!`. That means that we would also keep the four locations.